pennylane/utils.py

```diff
@@ -353,3 +356,30 @@ def __str__(self):
                 output += "{}({}(wires={}))\n".format(return_map[op.return_type], op.name, op.wires)
 
         return output
+
+
+def inv(operation_list):
+    """Invert a list of operations or a template.
+
+    If the inversion happens inside a QNode, the operations are removed and requeued
+    in the reversed order for proper inversion.
+
+    Args:
+        operation_list (Sequence[~.Operation]): A sequence of operations
+
+    Returns:
+        Sequence[~.Operation]: The inverted sequence of operations
+    """
+    inv_ops = [op.inv() for op in reversed(copy.deepcopy(operation_list))]
```

Comment:
What would be the reason for the need of the ``deepcopy`` here? The ``reversed`` operation seems to leave the original list intact:
```python
In [1]: a = [1,2] 
In [2]: b = list(reversed(a))                                                                                                                                                                               

In [3]: b                                                                                                                                                                                                   
Out[3]: [2, 1]

In [4]: a                                                                                                                                                                                                   
Out[4]: [1, 2]
```

Reply:
It leaves the original list intact, yes. But it copies the contents. We need the new inversion operation instances to have different IDs than the original ones, hence the deepcopy.

Reply:
That makes sense, thanks for that!